plugins/restful/RestfulDataProviderDbQuery.php

```diff
@@ -349,25 +349,43 @@ public function update($id, $full_replace = FALSE) {
     // Build the update array.
     $request = $this->getRequest();
     static::cleanRequest($request);
+    $save = FALSE;
+    $original_request = $request;
+
     $public_fields = $this->getPublicFields();
     $fields = array();
-    foreach ($public_fields as $public_property => $info) {
+    foreach ($public_fields as $public_field_name => $info) {
+      if (!empty($info['create_or_update_passthrough'])) {
+        // Allow passing the value in the request.
+        if (!empty($info['create_or_update_passthrough_required']) && !isset($request[$public_field_name])) {
+          throw new \RestfulBadRequestException(format_string('Property @name is required.', array('@name' => $public_field_name)));
+        }
+
+        unset($original_request[$public_field_name]);
+        continue;
+      }
+
       // If this is the primary field, skip.
       if ($this->isPrimaryField($info['property'])) {
         continue;
       }
       // Check if the public property is set in the payload.
-      if (!isset($request[$public_property])) {
+      if (!isset($request[$public_field_name])) {
         if ($full_replace) {
           $fields[$info['property']] = NULL;
         }
       }
       else {
-        $fields[$info['property']] = $request[$public_property];
+        $fields[$info['property']] = $request[$public_field_name];
       }
+
+      unset($original_request[$public_field_name]);
```

Comment:
this is a a bad copy/ paste - in the RestfulEntityBase we use $original_request to make sure all properties were mapped properly before saving them. I'll fix this